lib/ecto/adapters/mysql/connection.ex

```diff
@@ -495,7 +495,7 @@ if Code.ensure_loaded?(Mariaex) do
         case elem(sources, pos) do
           {table, schema} ->
             name = String.first(table) <> Integer.to_string(pos)
-            {quote_table(prefix, table), name, schema}
+            {quote_table(prefix || (schema && schema.__schema__(:prefix)), table), name, schema}
```

Comment:
Generally we avoid calling schema functions in the adapter. The idea is that the adapter already receives all the data pre-processed. This would also break cases where the prefix is explicitly set to nil.

Is there any reason why you preferred to change it here instead of setting in the ecto/association.ex file when the query is built as previously mentioned?